src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -391,42 +391,79 @@ public Set<DatasetDTO> describeDatasets(Integer userId) {
         List<Integer> datasetIdsInUse = dataAccessRequestDAO.findAllDARDatasetRelationDatasetIds();
         HashSet<DatasetDTO> datasets = new HashSet<>();
         if (userHasRole(UserRoles.ADMIN.getRoleName(), userId)) {
-            datasets.addAll(datasetDAO.findAllDatasets());
+            datasets.addAll(datasetDAO.findAllDatasetDTOs());
         } else {
-            datasets.addAll(datasetDAO.findActiveDatasets());
+            datasets.addAll(datasetDAO.findActiveDatasetDTOs());
             if (userHasRole(UserRoles.CHAIRPERSON.getRoleName(), userId)) {
-                Collection<DatasetDTO> chairSpecificDatasets = datasetDAO.findDatasetsByUserId(userId);
+                Collection<DatasetDTO> chairSpecificDatasets = datasetDAO.findDatasetDTOsByUserId(userId);
                 datasets.addAll(chairSpecificDatasets);
             }
         }
         datasets.forEach(d -> d.setDeletable(!datasetIdsInUse.contains(d.getDataSetId())));
         return datasets;
     }
 
+
+    public List<Dataset> searchDatasets(String query, User user) {
+        List<Dataset> datasets = this.findAllDatasetsByUser(user);
+
+        return datasets.stream().filter(ds ->this.filterDatasetByQuery(ds, query)).toList();
+    }
+
+
+    // dataset properties which can be searched by
+    private static final List<String> filterableDatasetProperties = List.of(
+            "Principal Investigator(PI)"
+    );
+    private boolean filterDatasetByQuery(Dataset ds, String query) {
+        List<String> terms = new ArrayList<>();
+
+        terms.add(ds.getName());
+        terms.add(ds.getDatasetIdentifier());
+        terms.add(ds.getDatasetName());
+
+        if (Objects.nonNull(ds.getProperties())) {
+            ds.getProperties().forEach(prop -> {
+                if (Objects.nonNull(prop.getPropertyName()) && Objects.nonNull(prop.getPropertyValue())
+                        && filterableDatasetProperties.contains(prop.getPropertyName())) {
+                    terms.add(prop.getPropertyValueAsString());
+                }
+            });
+        }
```

Comment:
If we're expanding the search to any field, we can simplify and speed this up a bit. Stream over all props for non-null values, map to lower case, collect to a list, then add the names and identifier after the list is constructed.

Reply:
Another thought here is that you could extract this logic to `Dataset` itself, i.e. 
```
public boolean isStringMatch(String query) { ... }
```
which would make streaming over a list of datasets a lot cleaner here in the service class.